src/Glimpse.Server.Web/Storage/InMemoryStorage.cs

```diff
@@ -8,65 +7,200 @@ namespace Glimpse.Server.Web
 {
     public class InMemoryStorage : IStorage, IQueryRequests
     {
+        /// <summary>
+        /// Internal class to contain all data associated with a single request. 
+        /// </summary>
+        private class RequestInfo
+        {
+            public RequestInfo(LinkedListNode<Guid> requestIdNode)
+            {
+                this.RequestIdNode = requestIdNode;
+                this.Messages = new LinkedList<IMessage>();
+                this.Indices = null;
+            }
+
+            /// <summary>
+            /// List of individual messages associated with this request
+            /// </summary>
+            public LinkedList<IMessage> Messages { get; private set; }
+            
+            /// <summary>
+            /// The node in the _activeRequests list.  Storing this here allows us to determine to track 
+            /// requests by age in constant time.
+            /// </summary>
+            public LinkedListNode<Guid> RequestIdNode { get; private set; }
```

Comment:
Note that these are public methods on an private inner class, so external callers won’t have access.